flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/sink/shuffle/TestDataStatisticsOperator.java

```diff
@@ -119,9 +121,9 @@ public void testProcessElement() throws Exception {
         testHarness = createHarness(this.operator)) {
       StateInitializationContext stateContext = getStateContext();
       operator.initializeState(stateContext);
-      operator.processElement(new StreamRecord<>(GenericRowData.of(StringData.fromString("a"))));
-      operator.processElement(new StreamRecord<>(GenericRowData.of(StringData.fromString("a"))));
-      operator.processElement(new StreamRecord<>(GenericRowData.of(StringData.fromString("b"))));
+      operator.processElement(new StreamRecord<>(genericRowDataA));
+      operator.processElement(new StreamRecord<>(genericRowDataA));
```

Comment:
Nit: could we make something like this here:
```
genericRowDataA_2 = GenericRowData.of(StringData.fromString("a"), 2);
operator.processElement(new StreamRecord<>(genericRowDataA_2));
```

Reply:
do you mean not reusing the object? if yes, I would just get rid of the pre-constructed objects and always construct them on the fly.

Reply:
Sorry - my midnight review was not clear enough. 😢 

I was trying to suggest to process a different `RowData` object, with the same key, like:
```
private final GenericRowData genericRowDataA_1 = GenericRowData.of(StringData.fromString("a"), 1);
private final GenericRowData genericRowDataA_2 = GenericRowData.of(StringData.fromString("a"), 2);
private final GenericRowData genericRowDataB = GenericRowData.of(StringData.fromString("b"), 3);
[..]
operator.processElement(new StreamRecord<>(genericRowDataA_1));
operator.processElement(new StreamRecord<>(genericRowDataA_2));
operator.processElement(new StreamRecord<>(genericRowDataB));
```

I know that we have individual test for the correct grouping, but I consider this as an e2e tests, and it would be nice to test this out as well.

Reply:
got it. for readability,I would just get rid of the pre-constructed objects and always construct them on the fly then